### PR TITLE
Deploy log analytics in same resource group as workspace in basic scenario

### PR DESCRIPTION
**Pull request by joshharrin**
# Description

Previous changes related to the move to log analytics-based app insights is less ideal for beginning customers or customers with additional requirements. Move scenario when user does not specify existing app insights to create a new log analytics resource per workspace in the same resource group as the workspace. User can decide on their own whether to reuse those resources or not and be responsible for the management themselves instead of creating/sharing in default resource groups. This adds confusion for the customer and can cause issues when user does not have perms to deploy a resource group.

This PR also improved messages output to user while creating workspace. 

Addresses following bugs:
https://dev.azure.com/msdata/Vienna/_workitems/edit/2287026/
https://dev.azure.com/msdata/Vienna/_workitems/edit/2285497/

## New experience:
az ml workspace create -n wsnoappinsightsspecified -g joharrin-sdk-tests
image
log analytics to be created in same rg as workspace
image

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by mingweihe — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json [hunk 1]
```diff
@@ -128,6 +128,20 @@
                 "description": "Determines whether or not new ApplicationInsights should be provisioned."
             }
         },
+        "logAnalyticsName": {
```
Comment (seanyao1) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json:131]: why do we need both logAnalyticsName and LogAnalyticsArmId?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_param.json [hunk 2]
```diff
@@ -44,6 +44,12 @@
     "applicationInsightsName": {
         "value": ""
     },
+    "logAnalyticsName": {
```
Comment (seanyao1) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_param.json:47]: same as above

File: sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json (hunk 1 above)
Comment (joshharrin) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json:131]: log analytics name is used for naming the resource to be created while log analytics arm id is used in the creation of the app insights resource

File: sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_param.json (hunk 2 above)
Comment (joshharrin) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_param.json:47]: commented above, name is used for naming resource, arm id is used for (log analytics) workspaceId in app insights creation

**Pull request merged**